test/e2e/specs/specs-playwright/wp-signup__wpcc.ts

```diff
@@ -0,0 +1,139 @@
+				const userSignupPage = new UserSignupPage( page );
+				await userSignupPage.signupWPCC( email, signupPassword );
+			} );
+
+			it( 'User lands in CrowdSignal dashboard', async function () {
+				// This will be a production site instead of staging or wpcalypso.
+				await page.waitForSelector( 'div.welcome-main' );
+			} );
+
+			it( 'Get activation link', async function () {
+				const emailClient = new EmailClient();
+				const message = await emailClient.getLastEmail( {
+					inboxId: inboxId,
+					emailAddress: email,
+					subject: 'Activate',
+				} );
+				const links = await emailClient.getLinksFromMessage( message );
+				activationLink = links.find( ( link: string ) => link.includes( 'activate' ) ) as string;
+			} );
+
+			it( 'Activate account', async function () {
+				await page.goto( activationLink );
+			} );
+		} );
+
+		describe( 'Follow site with Reader', function () {
+			let readerPage: ReaderPage;
+
+			it( 'Navigate to WordPress.com', async function () {
+				// Return to the webapp under test (staging/wpcalypso).
+				await page.goto( DataHelper.getCalypsoURL() );
+			} );
+
+			it( 'Land in Reader and search', async function () {
+				readerPage = new ReaderPage( page );
+				await readerPage.search( 'e2eflowtestingnotifications' );
+			} );
+
+			it( 'Read first post', async function () {
+				await readerPage.visitPost( { index: 1 } );
+			} );
+
+			it( 'Follow site', async function () {
+				await readerPage.followSite();
+			} );
+
+			it( 'Navigate back to search listing', async function () {
+				await page.goBack();
+			} );
+		} );
+
+		describe( 'Support: Show me where', function () {
+			let supportComponent: SupportComponent;
+
+			it( 'Open Support popover', async function () {
+				supportComponent = new SupportComponent( page );
+				await supportComponent.openPopover();
+			} );
+
+			it( 'Search for help: Create a site', async function () {
+				await supportComponent.search( 'create a site' );
+			} );
+
+			it( 'Click on result under Show me where', async function () {
+				await supportComponent.clickResult( 'where', 1 );
+			} );
+
+			it( 'Exit Gutenboarding flow', async function () {
+				const gutenboardingFlow = new GutenboardingFlow( page );
+				await gutenboardingFlow.clickWpLogo();
+			} );
+		} );
+
+		describe( 'Delete user account and ensure no longer accessible', function () {
+			let loginPage: LoginPage;
+
+			it( 'Close account', async function () {
+				const closeAccountFlow = new CloseAccountFlow( page );
+				await closeAccountFlow.closeAccount();
+			} );
+
+			it( 'Navigate to WordPress.com Login page', async function () {
+				loginPage = new LoginPage( page );
+				await loginPage.visit();
+			} );
+
+			it( 'Ensure user is unable to log in', async function () {
+				await loginPage.login( { username: email, password: signupPassword } );
+				await page.waitForSelector( ':text("This account has been closed")' );
+			} );
+		} );
```

Comment:
I've noticed that the setup (create username & password) and teardown (close account) steps are almost identical for every signup suite (free/paid/wpcc). IMO, it would be good to make those shareable between the suites for better consistency. For example, currently, only the wpcc suite's account closing is followed by ensuring that the account is closed. What do you think? cc @dpasque

Reply:
This following basic flow is indeed shared across all Signup tests:

```

			it( 'Close account', async function () {
				const closeAccountFlow = new CloseAccountFlow( page );
				await closeAccountFlow.closeAccount();
			} );
```

The two other steps are specific to WPCC Signup in that I am taking additional steps to verify the account closure indeed took place. In fact the `'Ensure user is unable to log in'` step does no exist anywhere else.

Reply:
As for the setup, I personally think it's better form to set up the values the test will use in the test itself instead of moving them behind a 'shared step' or something similar. Having a 'shared step' makes it difficult to follow the flow in my opinion.

Perhaps it would make sense to have another method in `DataHelper` which returns a generated username/password pair? That would keep the explicitness while removing one line of unnecessary setup.

Reply:
> As for the setup, I personally think it's better form to set up the values the test will use in the test itself instead of moving them behind a 'shared step' or something similar. Having a 'shared step' makes it difficult to follow the flow in my opinion.

Makes sense, let's not share steps 👍 

> Perhaps it would make sense to have another method in DataHelper which returns a generated email/password pair?

The `getNewUserCredentials` helper sounds good to me - would be nice to have such a wrapper 👍

Reply:
@WunderBart Mind if this change is proposed in a separate PR?

Reply:
Absolutely 👍